front_end/src/app/services/phoneline.service.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Phoneline } from '../models/phoneline';
+import { BehaviorSubject } from 'rxjs/internal/BehaviorSubject';
+import { Customer } from '../models/customer';
+import { Device } from '../models/device';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PhonelineService {
+
+  localHost: string = 'http://localhost:8080/customers';
+
+  phonelinesRaw: Phoneline[] = [];
+  phonelinesSubject = new BehaviorSubject<Phoneline[]>([]);
+  allPhonelines = this.phonelinesSubject.asObservable();
+
+  constructor(private http: HttpClient) {}
+
+  getAllPhonelines(custId: string) {
+    this.http.get<any>(`${this.localHost}/${custId}/lines`, { observe: 'response' })
+    .subscribe(data => {
+
+      this.phonelinesRaw = [];
+      console.log(data);
+      for (let phoneline of data.body) {
+        this.phonelinesRaw.push(new Phoneline(
+          phoneline.phoneNumber,
+          new Device(phoneline.deviceResponseDto.deviceId, phoneline.deviceResponseDto.name, phoneline.deviceResponseDto.brand, phoneline.deviceResponseDto.price)
+        ));
+      }
+      this.phonelinesSubject.next(this.phonelinesRaw);
+    });
+  }
+
+  createPhoneline(custId: string, phoneline: Phoneline) {
+    this.http.post<any>(`${this.localHost}/${custId}/lines`, phoneline, { observe: 'response' })
+    .subscribe(data => {
```

Comment:
This call is not quite working right. @parksamuelsoung Can you help debug this Wednesday?

Reply:
yes